test/integration/query-interface/changeColumn.test.js

```diff
@@ -361,6 +362,131 @@ describe(Support.getTestDialectTeaser('QueryInterface'), () => {
         expect(refs[0].referencedTableName).to.equal('Users');
         expect(refs[0].referencedColumnName).to.equal('id');
       });
+
+      it('should retain ON UPDATE and ON DELETE constraints after a column is changed', async function () {
+        await this.queryInterface.createTable('users', {
+          id: {
+            type: DataTypes.INTEGER,
+            primaryKey: true,
+            autoIncrement: true,
+          },
+          name: {
+            type: DataTypes.STRING,
+            allowNull: true,
+          },
+          level_id: {
+            type: DataTypes.INTEGER,
+            allowNull: false,
+            references: {
+              key: 'id',
+              model: 'level',
+            },
+            onDelete: 'CASCADE',
+            onUpdate: 'CASCADE',
+          },
+        });
+
+        await this.queryInterface.createTable('level', {
+          id: {
+            type: DataTypes.INTEGER,
+            primaryKey: true,
+            autoIncrement: true,
+          },
+          name: {
+            type: DataTypes.CHAR,
+            allowNull: false,
+          },
+        });
+
+        await this.queryInterface.changeColumn('users', 'name', {
+          type: DataTypes.CHAR,
+          allowNull: false,
+        });
+
+        const constraintsQuery = this.queryInterface.queryGenerator.showConstraintsQuery('users');
+        const [{ sql: usersSql }] = await this.queryInterface.sequelize.query(constraintsQuery, {
+          type: 'SELECT',
+        });
```

Comment:
Why not this?

```suggestion
        const constraintsQuery = this.queryInterface.showConstraint('users');
```

Reply:
I have tried the ```this.queryInterface.showConstraint('users');``` approach but it does not work for sqlite. It seems the problem lies within `src/dialects/abstract/query-interface.js#showConstraint` function. The `QueryTypes.SHOWCONSTRAINTS` type returns empty result for the sqlite dialect. This poses a problem however if we are to support all dialects within these new tests. 
I have problems with porting these test cases for MariaDB and MySQL (possibly others) due to using the workaround for sqlite (the one used in the test case before the suggested change). If you have a workaround or a suggestion I would gladly apply it.

Reply:
Would you prefer (1) a different test case for each dialect or (2) a branching logic within the same test depending on the dialect type? 

1. 
```js

if (dialect === 'sqlite) {
 it('sqlite specific version of this test', ...)
} else if (dialect === 'mysql') {
 it('mysql version of the test', ...)
}
...
```
2.
```js
it('some generic test', async () => {
  if (dialect === 'sqlite') {
    expect().some.condition()
  } else {
    expect().some.other.condition()
  }
}

```

Reply:
Also there is a 3rd solution, to fix the `showConstraint` functionality for sqlite dialect altough this would be out of the scope of this PR.

Reply:
Definitely option 2. However I would also prefer to fix `showConstraint` first, maybe in a parallel PR we can merge first? (if you're willing to help with that too)

Reply:
Yup, sure. I will open a new PR with a fix for `showConstraint`. Can you point me to a type definition for a valid return type for this function?

Reply:
I ran this locally and I concur that fixing sqlite's showConstraint is going to be very complicated and I don't want to block this PR until that minor issue is fixed

Can you add the following comment above the call to `showConstraintsQuery`?

`// TODO: replace with queryInterface.showConstraint once it lists foreign keys properly for sqlite`

Reply:
Alright. I have added a comment as per your suggestion.